File: php/send/ConsoleMessageSender.php
<?php

require_once 'IMessageSender.php';

class ConsoleMessageSender implements IMessageSender {

	public function send(array $messages) : void {
		printf("| Got %s messages:\n", count($messages));
		foreach ($messages as $message) {
			printf("| >%s\n", $message);
		}
		printf("| DONE\n");
	}

}

?>

File: java/src/main/java/com/github/yuvallb/sendEventDemo/send/ConsoleMessageSender.java
package com.github.yuvallb.sendEventDemo.send;

import java.util.List;

public class ConsoleMessageSender implements IMessageSender {

	public void send(List<String> messages) {
		if (messages == null) {
			System.out.println("| Got null message");
			return;
		}
		System.out.println(String.format("| Got %d messages:", messages.size()));
		for (String message: messages) {
			System.out.println("|  >  " + message);
		}
		System.out.println("| DONE");
	}

}
